### DISCUSSION TITLE
Bug in color output: Actions UI removes whitespace

### DISCUSSION THREAD

**Discussion by kachkaev** [community/community#18492] [Actions] [answered]
👋 I am using colored output in GitHub actions and have noticed a new bug in the UI. Here’s a repro: https://github.com/kachkaev/github-actions-color-ouput-bug/runs/6843618944?check_suite_focus=true#step:2:5

<img width="515" alt="Screenshot 2022-06-13 at 22 46 32" src="https://user-images.githubusercontent.com/608862/173450807-9c880dd3-0f6d-4cfb-8fcd-030c57e16ec5.png">

Running `echo -e "\e[31mRed Text\e[0m \e[32mGreen Text\e[0m"` in a local terminal produces `Red Text Green Text`, but GitHub actions UI squashes the whitespace, so we see `Red TextGreen Text`.

This appears to be a CSS / HTML issue in the logs. Page markup contains this:

```
<span class="ansifg-r" style=""><!----><!----><!----><!----><!---->Red Text<!----><!----><!----></span>
<span class="" style=""><!----><!----></span>
<span class="ansifg-g" style=""><!----><!----><!----><!----><!---->Green Text<!----><!----><!----></span>
```

Replacing `<span class="" style=""><!----><!----></span>` with `<span class="" style="">[space]</span>` fixes the representation:

<img width="527" alt="Screenshot 2022-06-14 at 21 41 38" src="https://user-images.githubusercontent.com/608862/173684806-a2addaa4-b4e2-4cd3-bb11-72905fd80f8e.png">

Please let me know where else to report the issue if this is the wrong place! Either way, it’d be great if someone could help me route this request to a corresponding team. I hope that the bugfix will be simple! 🤞

**PS:** I initially posted this report [on github.community](https://github.community/t/bug-in-color-output-actions-ui-removes-whitespace/256040) and @Akash1134 asked me to move it here 🙂

**Comment by MathiasReker** [comment]
A hotfix could be to use another whitespace charactor: https://qwerty.dev/whitespace/

**Reply by soliton-** [reply]
I noticed the same issue: https://github.com/soliton-/colortest/runs/7212373724?check_suite_focus=true#step:2:175

Unfortunate when printing JSON in color with jq for example. You can see in the log I also tried replacing spaces with several other space characters but it didn't help.

**Comment by kachkaev** [comment, marked as answer]
Seems like the issue is now fixed 🎉

https://github.com/kachkaev/github-actions-color-ouput-bug/actions/runs/4839891538/jobs/8625203087
<img width="507" alt="Screenshot 2023-04-29 at 18 11 59" src="https://user-images.githubusercontent.com/608862/235315143-96689d78-08b3-41ed-9da1-09b39be970b9.png">
